app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+# : true
+
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def last_five_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def increment_counter
+    author.increment!(:post_counter)
+  end
```

Comment:
[OPTIONAL] It would be nice if you used one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private.

- You can apply the same logic for the `after_save` callback with the `update_comments_counter` and `update_likes_counter` too.

Reply:
Dear @KDlamini, The suggested fix has been implemented.

Reply:
- [ ] [optional] @chelobotix great work implementing the changes requested, it would be even better if you used one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the #increment_counter method every time a new post has been saved to the database, then make the increment_counter method private like you did for post and comment models.